modules/Bio/EnsEMBL/Compara/RunnableDB/HomologyAnnotation/BlastFactory.pm

```diff
@@ -44,32 +42,33 @@ sub param_defaults {
     my $self = shift;
     return {
         %{$self->SUPER::param_defaults},
+        'step' => 200,
     };
 }
 
 
 sub fetch_input {
     my $self = shift @_;
 
-    my $species_set_id = $self->param_required('species_set_id');
-    my $species_set    = $self->compara_dba->get_SpeciesSetAdaptor->fetch_by_dbID($species_set_id);
-    my $genome_dbs     = $species_set->genome_dbs;
-
-    my @all_members;
-    my @genome_db_ids;
+    my $genome_dbs = $self->compara_dba->get_GenomeDBAdaptor->fetch_all();
+    my ( @genome_db_ids, @query_members );
 
     foreach my $genome_db (@$genome_dbs) {
         my $genome_db_id = $genome_db->dbID;
-        push @genome_db_ids, $genome_db_id;
         my $some_members = $self->compara_dba->get_SeqMemberAdaptor->_fetch_all_representative_for_blast_by_genome_db_id($genome_db_id);
+
+        my @mlsss        = @{$self->compara_dba->get_MethodLinkSpeciesSetAdaptor->fetch_all_by_method_link_type_GenomeDB('ENSEMBL_HOMOLOGUES', $genome_db)};
+        my @genome_members;
+
         foreach my $member (@$some_members) {
             my $member_id = $member->dbID;
-            push @all_members, $member_id;
+            push @genome_members, $member_id;
         }
+
+        push @query_members, { 'genome_db_id' => $genome_db_id, 'mlss_id' => $mlsss[0]->dbID, 'member_ids' => \@genome_members };
```

Comment:
If you only expect one element in `$mlsss` you should raise an exception if more than one is returned beforehand.